lib/mdns_lite/responder.ex

```diff
@@ -101,9 +101,14 @@ defmodule MdnsLite.Responder do
     if !dns_record.header.qr && length(dns_record.qdlist) > 0 do
       # There can be multiple queries in each request
       dns_record.qdlist
-      |> Enum.map(fn qd -> Query.handle(qd, state) end)
-      |> Enum.each(fn resources ->
-        send_response(resources, dns_record, mdns_destination(src_ip, src_port), state)
+      |> Enum.map(fn qd -> {qd.class, Query.handle(qd, state)} end)
```

Comment:
I'm thinking that this would look nicer if it were:

```elixir
dns_record.qdlist
|> Enum.map(&extract_destination/1) # returns {dest_tuple, updated_qd}
|> Enum.map(fn {dest_tuple, qd} -> {dest_tuple, Query.handle(qd, state)} end)
|> Enum.each(fn {dest_tuple, resources} -> send_response(resources, dns_record, dest_tuple, state)
```